src/net.c

```diff
@@ -589,6 +588,13 @@ static void periodic_handler(event_loop_t *loop, void *data) {
 		logger(mesh, MESHLINK_DEBUG, "--- autoconnect end ---");
 	}
 
+	for splay_each(node_t, n, mesh->nodes) {
+		if(n->status.dirty) {
+			node_write_config(mesh, mesh->self);
```

Comment:
`node_write_config()` shouldn't be writing `mesh->self` here but the selected dirty node handle

Reply:
Fixed.